src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+                    {
+                        return;
+                    }
+
+                    if (guardMethods.IsEmpty || !(context.OperationBlocks.GetControlFlowGraph() is { } cfg))
+                    {
+                        ReportDiagnosticsForAll(platformSpecificOperations, context);
+                        return;
+                    }
+
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
+                    var analysisResult = GlobalFlowStateAnalysis.TryGetOrComputeResult(
+                        cfg, context.OwningSymbol, CreateOperationVisitor, wellKnownTypeProvider,
+                        context.Options, MinimumOsRule, performValueContentAnalysis: true,
+                        context.CancellationToken, out var valueContentAnalysisResult);
+
+                    if (analysisResult == null)
+                    {
+                        ReportDiagnosticsForAll(platformSpecificOperations, context);
+                        return;
+                    }
+
+                    foreach (var platformSpecificOperation in platformSpecificOperations)
+                    {
+                        var value = analysisResult[platformSpecificOperation.Key.Kind, platformSpecificOperation.Key.Syntax];
+                        PlatformAttributeInfo attribute = platformSpecificOperation.Value.FirstOrDefault();
+
+                        if (value.Kind == GlobalFlowStateAnalysisValueSetKind.Unknown)
+                        {
+                            if (platformSpecificOperation.Key.TryGetContainingLocalOrLambdaFunctionSymbol(out var containingSymbol))
+                            {
+                                var localResult = analysisResult.TryGetInterproceduralResultByDefinition(containingSymbol);
+                                if (localResult != null)
+                                {
+                                    var localValue = localResult[platformSpecificOperation.Key.Kind, platformSpecificOperation.Key.Syntax];
+                                    if (localValue.Kind == GlobalFlowStateAnalysisValueSetKind.Known)
+                                    {
+                                        if (IsKnownValueGuarded(attribute, localValue))
+                                        {
+                                            continue;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        else if (value.Kind == GlobalFlowStateAnalysisValueSetKind.Known)
+                        {
+                            if (IsKnownValueGuarded(attribute, value))
+                            {
+                                continue;
+                            }
+                        }
+
+                        ReportDiagnostics(platformSpecificOperation.Key, attribute, context);
```

Comment:
We should only report diagnostics when we know that `value.Kind == GlobalFlowStateAnalysisValueSetKind.Known` (or `localValue` for the other case). We should never report diagnostics with `Kind != GlobalFlowStateAnalysisValueSetKind.Known`, as that indicates flow analysis bailed out.

Reply:
Not really, not everything related to platform guard check, `Know` means there was some kind of platform check using one of the guard methods. But source code could not have any guard method or code section could not relate to any guard methods. At this point, `platformSpecificOperations` not have all possible operations, it has only operations attributed with any of the os platform attributes, here we only checking if it is guarded by with any guard methods (`IsOSPlatformOrLater` or `IsOSPlatformEarlierThan`)

Reply:
Sounds good - your claim stands until I can prove it with a test that shows otherwise!